### PR TITLE
Merge development into master

### PR DESCRIPTION
**Pull request by wvanbreukelen**
### Changes

This pull requests includes the following significant changes:
- Much better unit test integration of the UART library. Users can now include the UART library within their modules and inject a ```MockUART``` instance for simulating UART communication with the "real" hardware controller. 
- UART library users are able to use the ```hwlib::ostream``` and ```hwlib::istream``` methods to send a lot of datatypes.
- Improved the example code located within the ```main.cpp```
- Written basic UART lib tests.

### Makefiles

Note that to properly include the UART library within your module you need to change your ```Source.cmake``` and ```BuildModule.cmake```.

#### ```Source.cmake```
Add the following files to the source files: ```src/uart_connection.cpp```
and ```src/mock_uart.cpp```

Example:

```make
set (sources
    src/wrap-hwlib.cpp
    src/uart_connection.cpp
    src/mock_uart.cpp
)
```

#### ```BuildModule.cmake```
Add the following file to the source files: ```src/hardware_uart.cpp```

Example:

```make
set (sources ${sources}
    src/main.cpp
    src/wrap-hwlib.cpp
    src/libc-stub.cpp
    src/hardware_uart.cpp
)
```

### Code Example
Code example is available within the [main.cpp](https://github.com/R2D2-2018/UART_LIB/blob/development/src/main.cpp).

### CODE REVIEW

File: Sources.cmake [hunk 1]
```diff
@@ -26,7 +12,6 @@ include_directories (${catch}/single_include)
 
 set (sources
     src/wrap-hwlib.cpp
```
Comment (ghost) [inline Sources.cmake]: including hwlib in the Sources.cmake isn't needed anymore as it's already included in BuildModule.cmake and TestModule.cmake

File: test/test_main.cpp [hunk 2]
```diff
@@ -1,6 +1,26 @@
 #define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
 #include "catch.hpp"
+#include "../src/mock_uart.hpp"
 
-TEST_CASE( "Example Test Case" ) {
+TEST_CASE("Example Test Case") {
```
Comment (ghost) [inline test/test_main.cpp]: Remove example test case.

**Comment by wvanbreukelen** [comment]
Please do not merge yet, namespacing is addressed.

**Comment by wvanbreukelen** [comment]
@JasperSmienk You may now merge (if you approved my changes). @JulianvDoorn pointed out that the UART lib did not used namespacing. I've fixed this with a hotfix.

**Comment by ghost** [comment]
Looks good, merging.

**Pull request merged**